File: LinearSearch/LinearSearch.cpp
#include <bits/stdc++.h>
using namespace std;

int linearsearch(int arr[],int n,int key)
{
for(int i=0;i<n;i++)
{
  if(arr[i]==key)
  {
    return i;
  }
}
return -1;
}
int main() {
  int n;
  cout<<"enter the size of array: ";
  cin>>n;

  int arr[n],key,res_index;
  for(int i=0;i<n;i++)
  {
    cout<<"enter the element "<<i+1<<" in array: ";
    cin>>arr[i];
    cout<<endl;
  }
  cout<<"enter the key to search:";
  cin>>key;
  res_index=linearsearch(arr,n,key);
  if(res_index==-1)
  {
cout<<"\nelement "<<key<<" not found";
  }
  else
  {
    cout<<"\nelement "<<key<<" found at index "<<res_index;
  }


return 0;

}

File: LinearSearch/LinearSearch.java
import java.util.Scanner;

class Main{

   public int LinearSearch(int arr[],int n,int key)
  {
     for(int i=0;i<n;i++)
     {
       if(arr[i]==key)
          return i;
     }
      return -1;
  }

  public static void main(String[] args) 
  {
    Scanner sc=new Scanner(System.in);
    int n,key;
    System.out.print("enter the size of array: ");
    n=sc.nextInt();
    int arr[]=new int[n];
    for(int i=0;i<n;i++)
    {
      System.out.print("enter the element "+(i+1)+" :");
      arr[i]=sc.nextInt();
    }
    System.out.print("Enter the key to search: ");
    key=sc.nextInt();
    Main search=new Main();
    int res_index=search.LinearSearch(arr,n,key);
    if(res_index!=-1)
    {
      System.out.println("Key found in array at index "+res_index);
    }
    else
    System.out.println("key not found in array");
  }
}